Gateway/Response/PaymentCaptureDetailsHandler.php

```diff
@@ -38,22 +52,65 @@ public function handle(array $handlingSubject, array $response)
         /** @var OrderPaymentInterface $payment */
         $payment = $payment->getPayment();
 
-        // set pspReference as lastTransId only!
-        $payment->setLastTransId($response['pspReference']);
-
-        // The capture request will return a capture-received message, but it doesn't mean the capture has been final
-        // so the invoice is set to Pending
-        // The capture request will return a capture-received message, but it doesn't mean the capture has been final
-        // so the invoice is set to Pending
-        if ($response["response"] === '[capture-received]') {
-            $payment->setIsTransactionPending(true);
-
-            /**
-             * close current transaction because you have capture the goods
-             * but do not close the authorisation because you can still cancel/refund order
-             */
-            $payment->setIsTransactionClosed(true);
-            $payment->setShouldCloseParentTransaction(false);
+        if (array_key_exists(TransactionCapture::MULTIPLE_AUTHORIZATIONS, $response)) {
+            $this->handleMultipleCaptureRequests($payment, $response);
+        } else {
+            // set pspReference as lastTransId only!
+            $payment->setLastTransId($response['pspReference']);
+
+            // The capture request will return a capture-received message, but it doesn't mean the capture has been final
+            // so the invoice is set to Pending
+            if ($response["response"] === '[capture-received]') {
```

Comment:
Can we move `[capture-received]` to a constant? I see it is being used multiple times in 2 classes now

Reply:
bee1dc6